scaffold-example.js

```diff
@@ -0,0 +1,48 @@
+const fs = require('fs');
+
+const path = process.argv[2];
+
+if (! path || path === '') {
+  console.log('You must provide a path for the code example');
+  console.log('> node scaffold-example.js <path> <example name>');
+  process.exit(1);
+}
+
+if (! fs.existsSync(path)) {
+  console.log(`Path ${path} does not exists`);
+  process.exit(1);
+}
+
+const exampleName = process.argv[3];
+
+if (! exampleName || exampleName === '') {
+  console.log('You must provide an example name for the code example');
+  console.log('> node scaffold-example.js <path> <example name>');
+  process.exit(1);
+}
+
+
+
+
+['js', 'go', 'cpp', 'java'].forEach(language => {
+  fs.writeFile(`${path}/${exampleName}.${language}`, '//@todo', error => {
+    if (error) {
+      console.log(`Error for ${language} example: ${error.message}`);
+    }
+  });
+})
+
+const yamlContent = `name: ${exampleName}
+description:
+hooks:
+  before:
+  after:
+template: default
+expect: something
+`;
+
+fs.writeFile(`${path}/${exampleName}.yml`, yamlContent, error => {
+  if (error) {
+    console.log(`Error for yml config: ${error.message}`);
+  }
+});
```

Comment:
is there a chance this script can exit before the writeFile callbacks are all run?

Reply:
I am using sync version now :)